src/status_im/ethereum/transactions/core.cljs

```diff
@@ -119,10 +119,14 @@
                 watched-transactions))))
 
 (fx/defn add-transfer
+  "We determine a unique id for the transfer before adding it because some
+   transaction can contain multiple transfers and they would overwrite each other
+   in the transfer map if identified by hash"
   [{:keys [db] :as cofx} {:keys [hash id] :as transfer} address]
-  (let [transfer-by-hash (get-in db [:wallet :accounts address :transactions hash])]
-        ;;transfer-by-id   (get-in db [:wallet :transaction id]) ;; TODO didn't found any usage of this
-    (when-let [unique-id (when (not= transfer transfer-by-hash) ;(or transfer-by-id)
+  (let [transfer-by-hash (get-in db [:wallet :accounts address :transactions hash])
+        transfer-by-id   (get-in db [:wallet :transaction id])]
```

Comment:
@siphiuel @yenda could you please point where we set this? i couldn't find

Reply:
it's status-go side

Reply:
wut? i mean where in status react we assoc value to :wallet :transaction id`

Reply:
it's right here. the transfer has a hash and id when it arrives from status-go, and in this function we check if there is already a transfer in the map that has this hash or id as key, if the hash is already there we use the id, if the id is already there we already have the transfer in the map

Reply:
still there should be a place where we put `:transaction` i can't find it, to get `(get-in db [:wallet :transaction id])` we need to `put` it there first

Reply:
that is suppose to be the list of transactions